pkg/scheduler/queue/queue.go

```diff
@@ -25,19 +27,38 @@ const (
 )
 
 var (
-	ErrInvalidTenantID     = errors.New("invalid tenant id")
-	ErrTooManyRequests     = errors.New("too many outstanding requests")
-	ErrStopped             = errors.New("queue is stopped")
-	ErrQuerierShuttingDown = errors.New("querier has informed the scheduler it is shutting down")
+	ErrInvalidTenantID        = errors.New("invalid tenant id")
+	ErrTooManyRequests        = errors.New("too many outstanding requests")
+	ErrStopped                = errors.New("queue is stopped")
+	ErrQuerierShuttingDown    = errors.New("querier has informed the scheduler it is shutting down")
+	ErrTooManyQueueDimensions = errors.New("too many additional queue dimensions provided")
 )
 
+type SchedulerRequest struct {
```

Comment:
moved here to avoid import cycles.

Also this pattern seems more appropriate - it's a type for placing stuff in the scheduler queue, so it goes in `scheduler/queue`, and clients of the scheduler queue have to import the type in order to pass something into the scheduler queue - as in, it's the interface client's job to put the request in the shape the interface provider understands.